Directory.Build.props

```diff
@@ -24,5 +24,6 @@
 	<PropertyGroup>
 		<LangVersion>11.0</LangVersion>
 		<ManagePackageVersionsCentrally>true</ManagePackageVersionsCentrally>
+		<WarningsAsErrors>CS8785</WarningsAsErrors>

```

Comment:
What is CS8785?

Reply:
Generator failed to generate source (want to make sure build fails if the sync generator fails). Should I comment in here?

Reply:
No that's fine, just new to me and wanted to be sure it made sense. I think we'd also notice it if we manually defined the interface methods instead of using `[GenerateSync]` to also fill those out.

```csharp
public interface IRequestAuthenticator {
  Task<ID2LPrincipal> AuthenticateAsync( HttpRequestMessage request );
  ID2LPrincipal Authenticate( HttpRequestMessage request );
}

internal sealed partial class RequestAuthenticator : IRequestAuthenticator {
  [GenerateSync]
  Task<ID2LPrincipal> IRequestAuthenticator.AuthenticateAsync(
    HttpRequestMessage request
  ) { ... }
```

In the event of a failure it would complain that `RequestAuthenticator` doesn't implement the interface. Don't know if @j3parker has a preference.

Reply:
Generating the sync interface for you certainly makes it easier to maintain... I dno, I don't really understand why that diagnostic isn't already an error. Like why would you be running a generator if you were cool with it crashing? lol

Reply:
We also insert `[Blocking]` on the generated stuff, etc. FYI

Like if we were having you specify them manually I think we'd also want to analyze them to make sure they matched what we wanted... and at that point it seems simpler to just generate.